lib/web.mlx

```diff
@@ -1,7 +1,8 @@
 module Info = struct
   let title = "Dune Developer Preview"
   let curl_with_sh url =
-    Format.sprintf "curl -fsSL %s | sh" url  |> JSX.string
+    Format.sprintf "curl --proto '=https' --tlsv1.2 -sSf %s | sh" url
+    |> JSX.string
```

Comment:
Can you explain the issue here? Is this to prevent old protocols? It seems like it would be nice to support TLSv1.3 as well?

Reply:
- `--proto '=https'`
  - Restricts the protocol to HTTPS only. The `=` ensures that only HTTPS is used; other protocols will not be attempted.
- `--tlsv1.2`
  - Specifies that TLS version 1.2 must be used for the connection. v3 is a better option, but many curl distributions do not yet have that functionality built into them.

Reply:
It's a safe approach that's now widely used in rustup and other tools: https://rustup.rs

Reply:
Thanks for the explanation and references, I have one more question:

> The = ensures that only HTTPS is used; other protocols will not be attempted.

What protocols could it possibly use if the URL already uses `https://`? h3?

Reply:
It just prevents downgrade attacks. It's basically not required, but it's much better than nothing.

Reply:
The redirect option is fairly useful for us if we want to redirect in the future without breaking the URL.

Reply:
Is there any problem with breaking? I don't think there is much problem in this kind of case.

Reply:
We had this already, where the instructions were posted to different places (e.g. Discuss, the preview website etc.) but then we changed the URL to the script and it broke all the instructions which then had to be updated (instead of adding a redirect which would have kept them working).

Reply:
OK! I just fixed.